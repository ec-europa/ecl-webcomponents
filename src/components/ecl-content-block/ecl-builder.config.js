const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, '../../../node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  scripts: [
    {
      entry: path.resolve(nodeModules, '@ecl/vanilla-component-content-block/content-block.js'),
      dest: path.resolve(outputFolder, 'scripts/ecl-content-block-vanilla.js'),
      options: {
        banner,
        moduleName: 'ECL',
        includePaths,
        sourceMap: false,
      },
    },
  ],
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-content-block-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-content-block-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-content-block-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-content-block-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
