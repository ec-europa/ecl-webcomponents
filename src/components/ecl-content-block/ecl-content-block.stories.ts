const getArgs = () => {
  return {
    title: 'Title',
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus gravida ipsum ut lorem cursus,
                  quis tincidunt sem viverra. Nunc vestibulum, mauris quis porta venenatis, justo odio commodo tellus`,
    metaPrimary:  '["PRIMARY META", "DD Month Year"]',
    metaSecondary: '[{"label": "2018/10/22", "icon": "calendar"}, {"label": "Luxembourg", "icon": "location"}]',
  };
};

const getArgTypes = () => {
  return {
    title: {
      type: { name: 'string'},
      description: 'Title of the content block',
    },
    description: {
      type: { name: 'string'},
      description: 'Description',
    },
    metaPrimary: {
      name: "Primary meta",
      type: { name: 'string'},
      description: 'Primary meta element',
    },
    metaSecondary: {
      name: 'Secondary meta',
      type: { name: 'string'},
      description: 'Secondary meta element',
    },
  };
};

export default {
  title: 'Components/content-block',
};

const Template = (args) => 
  `<ecl-content-block
    theme="${args.theme}"
    meta-primary='${args.metaPrimary}'
    meta-secondary='${args.metaSecondary}'
    has-title
    has-description
    has-lists
    has-labels
    has-links
    has-secondary-links
  >
    <ecl-label
      slot="labels"
      variant="highlight"
      theme="${args.theme}"
    >
      Highlight
    </ecl-label>
    <ecl-label
      slot="labels"
      variant="high"
      theme="${args.theme}"
    >
      High importance
    </ecl-label>
    <ecl-text tag="p" slot="description">${args.description}</ecl-tag>
    <ecl-link
      theme="${args.theme}"
      slot="title"
      path="/example.html"
      variant="standalone"
    >
      ${args.title}
    </ecl-link>
    <ecl-link
      slot="links"
      theme="${args.theme}"
      variant="standalone"
      path="/example.html"
    >
      Primary Link 1
    </ecl-link>
    <ecl-link
      slot="links"
      theme="${args.theme}"
      variant="standalone"
      path="/example.html"
    >
      Primary Link 2
    </ecl-link>
    <ecl-link
      slot="links"
      theme="${args.theme}"
      variant="standalone"
      path="/example.html"
    >
      Primary Link 3
    </ecl-link>
    <ecl-link
      slot="links"
      theme="${args.theme}"
      variant="standalone"
      path="/example.html"
    >
      Primary Link 4
    </ecl-link>
    <ecl-link
      slot="links-secondary"
      theme="${args.theme}"
      variant="standalone"
      path="/example.html"
    >
      Secondary Link 1
    </ecl-link>
    <ecl-link
      slot="links-secondary"
      theme="${args.theme}"
      variant="standalone"
      path="/example.html"
    >
      Secondary Link 2
    </ecl-link>
    <ecl-description-list
      theme="${args.theme}"
      slot="lists"
    >
      <ecl-description-list-term
        theme="${args.theme}"
      >
        Standard text
      </ecl-description-list-term>
      <ecl-description-list-definition
        type="text"
        theme="${args.theme}"
      >
        Lorem ipsum dolor sit amet, <a href="/component-library/example" class="ecl-link">consectetur adipiscing elit</a>. Suspendisse ut sapien condimentum, aliquet turpis sit amet, finibus purus. Donec porttitor iaculis felis ut dapibus. Sed blandit, massa ac suscipit facilisis
      </ecl-description-list-definition>
      <ecl-description-list-term
        theme="${args.theme}"
      >
        Standalone links
      </ecl-description-list-term>
      <ecl-description-list-definition
        type="link"
        theme="${args.theme}"
        items='[{"label": "Lorem ipsum dolor sit amet", "path": "/example.html", "icon": "copy"},{"label": "Lorem ipsum dolor sit amet", "path": "/example.html", "icon": "download"}]'
      >
      </ecl-description-list-definition>
      <ecl-description-list-term
        theme="${args.theme}"
      >
        Links inline
      </ecl-description-list-term>
      <ecl-description-list-definition
        type="inline"
        theme="${args.theme}"
        items='[{"label": "Lorem ipsum dolor sit amet", "path": "/example.html"},{"label": "Lorem ipsum dolor sit amet", "path": "/example.html"}]'
      >
      </ecl-description-list-definition>
      <ecl-description-list-term
        theme="${args.theme}"
      >
        Taxonomy list
      </ecl-description-list-term>
      <ecl-description-list-definition
        type="taxonomy"
        theme="${args.theme}"
        items='["Taxonomy item 1",{"label": "Taxonomy item 2", "path": "/example.html"},"Taxonomy item 3"]'
      ></ecl-description-list-definition>
    </ecl-description-list>
  </ecl-content-block>`;

export const ContentBlock = Template.bind({});
ContentBlock.storyName = 'default';
ContentBlock.args = getArgs();
ContentBlock.argTypes = getArgTypes();
