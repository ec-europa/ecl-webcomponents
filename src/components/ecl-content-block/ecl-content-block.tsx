import { Component, Prop, h, Element } from '@stencil/core';
import getAssetPath from "../../utils/assetPath";
declare const ECL: any;

@Component({
  tag: 'ecl-content-block',
  styleUrls: {
    ec: './build/styles/ecl-content-block-ec.css',
    eu: './build/styles/ecl-content-block-eu.css',
  },
  assetsDirs: ['build'],
  shadow: false,
  scoped: true,
})

export class EclContentBlock {
  @Element() el: HTMLElement;
  @Prop() styleClass: string = '';
  @Prop() theme: string = 'ec';
  @Prop() hasDescription: boolean;
  @Prop() eclScript: boolean = false;
  @Prop() hasTitle: boolean;
  @Prop() hasLabels: boolean;
  @Prop() hasLinks: boolean;
  @Prop() hasSecondaryLinks: boolean;
  @Prop() hasLists: boolean;
  @Prop() metaPrimary: string;
  @Prop() metaSecondary: string;

  getClass(): string {
    return [
      `ecl-content-block`,
      this.styleClass
    ].join(' ');
  }

  componentDidRender() {
    if (this.hasLabels) {
      const labelsContainer = this.el.querySelector('.ecl-content-block__label-container');
      const labels = this.el.querySelectorAll('.ecl-label');
      if (labels) {
        labelsContainer.innerHTML = '';
        labels.forEach((label) => {
          label.classList.add(`sc-ecl-content-block-${this.theme}`);
          const listEl = document.createElement('li');
          listEl.classList.add('ecl-content-block__label-item', `sc-ecl-content-block-${this.theme}`);
          listEl.appendChild(label);
          labelsContainer.appendChild(listEl);
        });
      }
    }
    if (this.hasLinks) {
      const linksContainers = this.el.querySelectorAll('.ecl-content-block__link-list');

      linksContainers.forEach((linkList) => {
        const thisLinks = linkList.querySelectorAll('.ecl-link');
        if (thisLinks) {
          linkList.innerHTML = '';
          thisLinks.forEach((link) => {
            link.classList.add(`sc-ecl-content-block-${this.theme}`);
            const listEl = document.createElement('li');
            listEl.classList.add('ecl-content-block__link-item', `sc-ecl-content-block-${this.theme}`);
            listEl.appendChild(link);
            linkList.appendChild(listEl);
          });
        }
      });
    }
    if (this.hasLists) {
      const list = this.el.querySelector('.ecl-content-block__list-container .ecl-description-list');
      if (list) {
        list.classList.add('ecl-content-block__list', `sc-ecl-content-block-${this.theme}`);
      }
    }

    if (this.eclScript) {
      // Load the ECL vanilla js if not already present.
      const src = getAssetPath('./build/scripts/ecl-content-block-vanilla.js');
      if (document.querySelector(`script[src="${src}"]`)) {
        document.querySelector(`script[src="${src}"]`).remove();
      }
      const script = document.createElement('script');
      script.src = src;
      script.onload = () => {
        const contentBlock = new ECL.ContentBlock(this.el);
        contentBlock.init();
      };

      document.body.appendChild(script);
    }
  }

  render() {
    const metaPrimaryArray = this.metaPrimary ? JSON.parse(this.metaPrimary) : '';
    const metaSecondaryArray = this.metaSecondary ? JSON.parse(this.metaSecondary) : '';
    return (
     <div class={this.getClass()}>
      { this.hasLabels ?
        <ul class="ecl-content-block__label-container">
          <slot name="labels"></slot>
        </ul> : ''
      }
      { metaPrimaryArray ?
        <ul class="ecl-content-block__primary-meta-container">
        { metaPrimaryArray.map((meta) => (
          <li class="ecl-content-block__primary-meta-item">
            {meta}
          </li>
          ))
        }  
        </ul> : '' 
      }
      { this.hasTitle ?
        <h1 class="ecl-content-block__title">
          <slot name="title"></slot>
        </h1> : ''
      }
      { this.hasDescription ?
        <div class="ecl-content-block__description">
          <slot name="description"></slot>
        </div> : ''
      }
      { metaSecondaryArray ?
        <ul class="ecl-content-block__secondary-meta-container">
        { metaSecondaryArray.map((meta) => (
          <li class="ecl-content-block__secondary-meta-item">
          { meta.icon ?
            <ecl-icon
              theme={this.theme}
              icon={meta.icon}
              size="s"
              style-class={`ecl-content-block__secondary-meta-icon sc-ecl-content-block-${this.theme}`}
            ></ecl-icon> : ''
          }
          { meta.label ? 
            <span class="ecl-content-block__secondary-meta-label">
              {meta.label}
            </span> : ''
          }
          </li>
          ))
        }  
        </ul> : '' 
      }
      { this.hasLinks ?
        <div class="ecl-content-block__link-container">
          <ul class="ecl-content-block__link-list">
            <slot name="links"></slot>
          </ul>
          { this.hasSecondaryLinks ?
          <ul class="ecl-content-block__link-list">
            <slot name="links-secondary"></slot>
          </ul> : ''
          }
        </div> : ''
      }
      { this.hasLists ?
        <div class="ecl-content-block__list-container">
          <slot name="lists"></slot>
        </div> : ''
      }
     </div> 
    )
  }
}
