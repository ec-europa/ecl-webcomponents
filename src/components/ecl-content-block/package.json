{
  "name": "@ecl/web-components-content-block",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.4.0",
  "description": "ECL Content Block",
  "main": "ecl-content-block.tsx",
  "private": true,
  "scripts": {
    "build:styles": "cross-env NODE_ENV=production ecl-builder styles",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.7.1",
    "@ecl/vanilla-component-content-block": "3.7.1",
    "@ecl/vanilla-component-description-list": "3.7.1",
    "@ecl/vanilla-component-icon": "3.7.1",
    "@ecl/vanilla-component-label": "3.7.1",
    "@ecl/vanilla-component-link": "3.7.1",
    "cross-env": "7.0.3"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}
