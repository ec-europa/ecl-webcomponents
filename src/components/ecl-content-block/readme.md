# ecl-content-block



<!-- Auto Generated Below -->


## Properties

| Property            | Attribute             | Description | Type      | Default     |
| ------------------- | --------------------- | ----------- | --------- | ----------- |
| `hasDescription`    | `has-description`     |             | `boolean` | `undefined` |
| `hasLabels`         | `has-labels`          |             | `boolean` | `undefined` |
| `hasLinks`          | `has-links`           |             | `boolean` | `undefined` |
| `hasLists`          | `has-lists`           |             | `boolean` | `undefined` |
| `hasSecondaryLinks` | `has-secondary-links` |             | `boolean` | `undefined` |
| `hasTitle`          | `has-title`           |             | `boolean` | `undefined` |
| `metaPrimary`       | `meta-primary`        |             | `string`  | `undefined` |
| `metaSecondary`     | `meta-secondary`      |             | `string`  | `undefined` |
| `styleClass`        | `style-class`         |             | `string`  | `''`        |
| `theme`             | `theme`               |             | `string`  | `'ec'`      |


## Dependencies

### Depends on

- [ecl-icon](../ecl-icon)

### Graph
```mermaid
graph TD;
  ecl-content-block --> ecl-icon
  style ecl-content-block fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
