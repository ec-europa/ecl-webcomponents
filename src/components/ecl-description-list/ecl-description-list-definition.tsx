import { Component, Prop, h } from '@stencil/core';

@Component({
  tag: 'ecl-description-list-definition',
  shadow: false,
})
export class EclDescriptionListDefinition {
  @Prop() type: string = 'text';
  @Prop() styleClass: string = '';
  @Prop() theme: string = 'ec';
  @Prop() items: string;

  getClass(): string {
    return [
      `ecl-description-list__definition`,
      `sc-ecl-description-list-${this.theme}`,
      `ecl-description-list__definition--${this.type}`,
      this.styleClass
    ].join(' ');
  }

  getMarkup = (itemsArray) => {
    if (this.type === 'link' && itemsArray) {
      return itemsArray.map((link) => (
        <ecl-link
          style-class={`ecl-description-list__definition-item sc-ecl-description-list-${this.theme}`}
          path={link.path}
        >
      { link.icon ? 
        <ecl-icon
          slot="icon-before"
          icon={link.icon}
        >
        </ecl-icon>
        : '' }
          {link.label}
        </ecl-link> 
        ));
    } else if (this.type === 'link' && !itemsArray) {
        return <slot></slot>;
    }

    if (this.type === 'inline' && itemsArray) {
      return itemsArray.map((inline) => (
        <ecl-link
          style-class={`ecl-description-list__definition-item sc-ecl-description-list-${this.theme}`}
          path={inline.path}
        >
          {inline.label}
        </ecl-link> 
        ));
    } else if (this.type === 'inline' && !itemsArray) {
      return <slot></slot>;
    }
    
    if (this.type === 'taxonomy' && itemsArray) {
      return itemsArray.map((taxonomy) => (
        <span
          class={`ecl-description-list__definition-item sc-ecl-description-list-${this.theme}`}
        >
        { taxonomy.path ? 
          <ecl-link
            path={taxonomy.path}
          >
            {taxonomy.label}
          </ecl-link> : taxonomy
        }
        </span> 
      ))
    } else if (this.type === 'taxonomy' && !itemsArray) {
      return <slot></slot>;
    }
    
    if (this.type === 'text') {
      return <slot></slot>
    }
  }

  render() {
    const itemsArray = this.items ? JSON.parse(this.items) : false;

    return (
      <dd class={this.getClass()}>
        {this.getMarkup(itemsArray)}
      </dd>
    )
  }
}