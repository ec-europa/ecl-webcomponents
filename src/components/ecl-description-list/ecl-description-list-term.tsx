import { Component, Prop, h } from '@stencil/core';

@Component({
  tag: 'ecl-description-list-term',
  shadow: false,
  scoped: false,
})
export class EclDescriptionListTerm {
  @Prop() styleClass: string = '';
  @Prop({ mutable: true }) theme: string;

  componentWillLoad() {
    this.theme = document.documentElement.getAttribute('data-ecl-theme') ?? (this.theme || 'ec');
  }

  getClass(): string {
    return [
      `ecl-description-list__term`,
      `sc-ecl-description-list-${this.theme}`,
      this.styleClass
    ].join(' ');
  }

  render() {
    return (
      <dt class={this.getClass()}>
        <slot></slot>
      </dt>
    );
  }
}