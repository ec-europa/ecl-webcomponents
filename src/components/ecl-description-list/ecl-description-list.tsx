import { Component, Prop, h, Element } from '@stencil/core';
import getAssetPath from "../../utils/assetPath";
declare const ECL: any;

@Component({
  tag: 'ecl-description-list',
  styleUrls: {
    ec: './build/styles/ecl-description-list-ec.css',
    eu: './build/styles/ecl-description-list-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})
export class EclDescriptionList {
  @Element() el: HTMLElement;
  @Prop() styleClass: string = '';
  @Prop() theme: string = 'ec';
  @Prop() variant: string = 'vertical';
  @Prop() visibleItems: number;
  @Prop() moreLabel: string;
  @Prop() eclScript: boolean = false;

  getClass(): string {
    return [
      `ecl-description-list`,
      `ecl-description-list--${this.variant}`,
      this.styleClass
    ].join(' ');
  }

  getAttrs() {
    const attrs = {
      'class': this.getClass(),
    };

    if (this.visibleItems) {
      attrs['data-ecl-description-list'] = true;
      attrs['data-ecl-description-list-more-label'] = this.moreLabel;
      attrs['data-ecl-description-list-visible-items'] = this.visibleItems;
    }

    return attrs;
  }

  componentDidRender() {
    if (this.eclScript) {
      const src = getAssetPath('./build/scripts/ecl-description-list-vanilla.js');
      if (document.querySelector(`script[src="${src}"]`)) {
        document.querySelector(`script[src="${src}"]`).remove();
      }
      const script = document.createElement('script');
      script.src = src;
      script.onload = () => {
        const descriptionList = new ECL.DescriptionList(this.el.firstElementChild);
        descriptionList.init();
        const seeMore = this.el.querySelectorAll('.ecl-description-list__see_more');
        if (seeMore[0]) {
          seeMore.forEach((more) => {
            more.classList.add(`sc-ecl-description-list-${this.theme}`);
          });
        }
      };
      document.body.appendChild(script);
    }
  }

  render() {
    return (
      <dl {...this.getAttrs()}>
        <slot></slot>
      </dl>
    );
  }
}
