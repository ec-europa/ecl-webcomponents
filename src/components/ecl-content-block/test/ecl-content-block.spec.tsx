import { newSpecPage } from '@stencil/core/testing';
import { EclContentBlock } from '../ecl-content-block';
import { EclLabel } from '../../ecl-label/ecl-label';
import { EclLink } from '../../ecl-link/ecl-link';
import { EclIcon } from '../../ecl-icon/ecl-icon';
import { EclDescriptionList } from '../../ecl-description-list/ecl-description-list';
import { EclDescriptionListTerm } from '../../ecl-description-list/ecl-description-list-term';
import { EclDescriptionListDefinition } from '../../ecl-description-list/ecl-description-list-definition';

describe('ecl-content-block', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclContentBlock, EclLink, EclIcon, EclDescriptionList, EclLabel, EclDescriptionListDefinition, EclDescriptionListTerm],
      html: `<ecl-content-block
        theme="ec"
        meta-primary='["PRIMARY META", "DD Month Year"]'
        meta-secondary='[{"label": "2018/10/22", "icon": "calendar"}, {"label": "Luxembourg", "icon": "location"}]'
        has-title
        has-description
        has-lists
        has-labels
        has-links
        has-secondary-links
        ecl-script
      >
        <ecl-label
          slot="labels"
          variant="highlight"
          theme="ec"
        >
          Highlight
        </ecl-label>
        <ecl-label
          slot="labels"
          variant="high"
          theme="ec"
        >
          High importance
        </ecl-label>
        <p slot="description">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus gravida ipsum ut lorem cursus,
          quis tincidunt sem viverra. Nunc vestibulum, mauris quis porta venenatis, justo odio commodo tellus
        </p>
        <ecl-link
          theme="ec"
          slot="title"
          path="/example.html"
          variant="standalone"
        >
          Title
        </ecl-link>
        <ecl-link
          slot="links"
          theme="ec"
          variant="standalone"
          path="/example.html"
        >
          Primary Link 1
        </ecl-link>
        <ecl-link
          slot="links"
          theme="ec"
          variant="standalone"
          path="/example.html"
        >
          Primary Link 2
        </ecl-link>
        <ecl-link
          slot="links"
          theme="ec"
          variant="standalone"
          path="/example.html"
        >
          Primary Link 3
        </ecl-link>
        <ecl-link
          slot="links"
          theme="ec"
          variant="standalone"
          path="/example.html"
        >
          Primary Link 4
        </ecl-link>
        <ecl-link
          slot="links-secondary"
          theme="ec"
          variant="standalone"
          path="/example.html"
        >
          Secondary Link 1
        </ecl-link>
        <ecl-link
          slot="links-secondary"
          theme="ec"
          variant="standalone"
          path="/example.html"
        >
          Secondary Link 2
        </ecl-link>
        <ecl-description-list
          theme="ec"
          slot="lists"
        >
          <ecl-description-list-term
            theme="ec"
          >
            Standard text
          </ecl-description-list-term>
          <ecl-description-list-definition
            type="text"
            theme="ec"
          >
            Lorem ipsum dolor sit amet, <a href="/component-library/example.html" class="ecl-link">consectetur adipiscing elit</a>. Suspendisse ut sapien condimentum, aliquet turpis sit amet, finibus purus. Donec porttitor iaculis felis ut dapibus. Sed blandit, massa ac suscipit facilisis
          </ecl-description-list-definition>
          <ecl-description-list-term
            theme="ec"
          >
            Standalone links
          </ecl-description-list-term>
          <ecl-description-list-definition
            type="link"
            theme="ec"
            items='[{"label": "Lorem ipsum dolor sit amet", "path": "/example.html", "icon": "copy"},{"label": "Lorem ipsum dolor sit amet", "path": "/example.html", "icon": "download"}]'
          >
          </ecl-description-list-definition>
          <ecl-description-list-term
            theme="ec"
          >
            Links inline
          </ecl-description-list-term>
          <ecl-description-list-definition
            type="inline"
            theme="ec"
            items='[{"label": "Lorem ipsum dolor sit amet", "path": "/example.html"},{"label": "Lorem ipsum dolor sit amet", "path": "/example.html"}]'
          >
          </ecl-description-list-definition>
          <ecl-description-list-term
            theme="ec"
          >
            Taxonomy list
          </ecl-description-list-term>
          <ecl-description-list-definition
            type="taxonomy"
            theme="ec"
            items='["Taxonomy item 1",{"label": "Taxonomy item 2", "path": "/example.html"},"Taxonomy item 3"]'
          ></ecl-description-list-definition>
        </ecl-description-list>
      </ecl-content-block>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
