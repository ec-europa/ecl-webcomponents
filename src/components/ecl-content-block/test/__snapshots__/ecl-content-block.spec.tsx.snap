// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-content-block renders 1`] = `
<ecl-content-block has-description="" has-labels="" has-links="" has-lists="" has-secondary-links="" has-title="" meta-primary="[&quot;PRIMARY META&quot;, &quot;DD Month Year&quot;]" meta-secondary="[{&quot;label&quot;: &quot;2018/10/22&quot;, &quot;icon&quot;: &quot;calendar&quot;}, {&quot;label&quot;: &quot;Luxembourg&quot;, &quot;icon&quot;: &quot;location&quot;}]" theme="ec">
  <!---->
  <div class="ecl-content-block">
    <ul class="ecl-content-block__label-container">
      <li class="ecl-content-block__label-item sc-ecl-content-block-ec">
        <span class="ecl-label ecl-label--highlight sc-ecl-content-block-ec">
          Highlight
        </span>
      </li>
      <li class="ecl-content-block__label-item sc-ecl-content-block-ec">
        <span class="ecl-label ecl-label--high sc-ecl-content-block-ec">
          High importance
        </span>
      </li>
    </ul>
    <ul class="ecl-content-block__primary-meta-container">
      <li class="ecl-content-block__primary-meta-item">
        PRIMARY META
      </li>
      <li class="ecl-content-block__primary-meta-item">
        DD Month Year
      </li>
    </ul>
    <h1 class="ecl-content-block__title">
      <ecl-link path="/example.html" slot="title" theme="ec" variant="standalone">
        <!---->
        <a class="ecl-link ecl-link--standalone" href="/example.html">
          Title
          <slot-fb name="icon-after"></slot-fb>
        </a>
      </ecl-link>
    </h1>
    <div class="ecl-content-block__description">
      <p slot="description">
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus gravida ipsum ut lorem cursus, quis tincidunt sem viverra. Nunc vestibulum, mauris quis porta venenatis, justo odio commodo tellus
      </p>
    </div>
    <ul class="ecl-content-block__secondary-meta-container">
      <li class="ecl-content-block__secondary-meta-item">
        <ecl-icon style-class="ecl-content-block__secondary-meta-icon sc-ecl-content-block-ec">
          <svg class="ecl-content-block__secondary-meta-icon ecl-icon ecl-icon--s sc-ecl-content-block-ec">
            <use xlink:href="/build/images/ec/icons.svg#calendar"></use>
          </svg>
        </ecl-icon>
        <span class="ecl-content-block__secondary-meta-label">
          2018/10/22
        </span>
      </li>
      <li class="ecl-content-block__secondary-meta-item">
        <ecl-icon style-class="ecl-content-block__secondary-meta-icon sc-ecl-content-block-ec">
          <svg class="ecl-content-block__secondary-meta-icon ecl-icon ecl-icon--s sc-ecl-content-block-ec">
            <use xlink:href="/build/images/ec/icons.svg#location"></use>
          </svg>
        </ecl-icon>
        <span class="ecl-content-block__secondary-meta-label">
          Luxembourg
        </span>
      </li>
    </ul>
    <div class="ecl-content-block__link-container">
      <ul class="ecl-content-block__link-list">
        <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
          <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
            Primary Link 1
            <slot-fb name="icon-after"></slot-fb>
          </a>
        </li>
        <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
          <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
            Primary Link 2
            <slot-fb name="icon-after"></slot-fb>
          </a>
        </li>
        <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
          <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
            Primary Link 3
            <slot-fb name="icon-after"></slot-fb>
          </a>
        </li>
        <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
          <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
            Primary Link 4
            <slot-fb name="icon-after"></slot-fb>
          </a>
        </li>
      </ul>
      <ul class="ecl-content-block__link-list">
        <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
          <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
            Secondary Link 1
            <slot-fb name="icon-after"></slot-fb>
          </a>
        </li>
        <li class="ecl-content-block__link-item sc-ecl-content-block-ec">
          <a class="ecl-link ecl-link--standalone sc-ecl-content-block-ec" href="/example.html">
            Secondary Link 2
            <slot-fb name="icon-after"></slot-fb>
          </a>
        </li>
      </ul>
    </div>
    <div class="ecl-content-block__list-container">
      <ecl-description-list slot="lists" theme="ec">
        <!---->
        <dl class="ecl-content-block__list ecl-description-list ecl-description-list--vertical sc-ecl-content-block-ec">
          <ecl-description-list-term theme="ec">
            <!---->
            <dt class="ecl-description-list__term sc-ecl-description-list-ec">
              Standard text
            </dt>
          </ecl-description-list-term>
          <ecl-description-list-definition theme="ec" type="text">
            <!---->
            <dd class="ecl-description-list__definition ecl-description-list__definition--text sc-ecl-description-list-ec">
              Lorem ipsum dolor sit amet,
              <a class="ecl-link" href="/component-library/example">
                consectetur adipiscing elit
              </a>
              . Suspendisse ut sapien condimentum, aliquet turpis sit amet, finibus purus. Donec porttitor iaculis felis ut dapibus. Sed blandit, massa ac suscipit facilisis
            </dd>
          </ecl-description-list-definition>
          <ecl-description-list-term theme="ec">
            <!---->
            <dt class="ecl-description-list__term sc-ecl-description-list-ec">
              Standalone links
            </dt>
          </ecl-description-list-term>
          <ecl-description-list-definition items="[{&quot;label&quot;: &quot;Lorem ipsum dolor sit amet&quot;, &quot;path&quot;: &quot;/example.html&quot;, &quot;icon&quot;: &quot;copy&quot;},{&quot;label&quot;: &quot;Lorem ipsum dolor sit amet&quot;, &quot;path&quot;: &quot;/example.html&quot;, &quot;icon&quot;: &quot;download&quot;}]" theme="ec" type="link">
            <!---->
            <dd class="ecl-description-list__definition ecl-description-list__definition--link sc-ecl-description-list-ec">
              <ecl-link style-class="ecl-description-list__definition-item sc-ecl-description-list-ec">
                <!---->
                <a class="ecl-description-list__definition-item ecl-link ecl-link--default ecl-link--icon ecl-link--icon-before sc-ecl-description-list-ec" href="/example.html">
                  <ecl-icon slot="icon-before">
                    <svg class="ecl-icon ecl-icon--xs ecl-link__icon ecl-link__icon--before">
                      <use xlink:href="/build/images/ec/icons.svg#copy"></use>
                    </svg>
                  </ecl-icon>
                  Lorem ipsum dolor sit amet
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </ecl-link>
              <ecl-link style-class="ecl-description-list__definition-item sc-ecl-description-list-ec">
                <!---->
                <a class="ecl-description-list__definition-item ecl-link ecl-link--default ecl-link--icon ecl-link--icon-before sc-ecl-description-list-ec" href="/example.html">
                  <ecl-icon slot="icon-before">
                    <svg class="ecl-icon ecl-icon--xs ecl-link__icon ecl-link__icon--before">
                      <use xlink:href="/build/images/ec/icons.svg#download"></use>
                    </svg>
                  </ecl-icon>
                  Lorem ipsum dolor sit amet
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </ecl-link>
            </dd>
          </ecl-description-list-definition>
          <ecl-description-list-term theme="ec">
            <!---->
            <dt class="ecl-description-list__term sc-ecl-description-list-ec">
              Links inline
            </dt>
          </ecl-description-list-term>
          <ecl-description-list-definition items="[{&quot;label&quot;: &quot;Lorem ipsum dolor sit amet&quot;, &quot;path&quot;: &quot;/example.html&quot;},{&quot;label&quot;: &quot;Lorem ipsum dolor sit amet&quot;, &quot;path&quot;: &quot;/example.html&quot;}]" theme="ec" type="inline">
            <!---->
            <dd class="ecl-description-list__definition ecl-description-list__definition--inline sc-ecl-description-list-ec">
              <ecl-link style-class="ecl-description-list__definition-item sc-ecl-description-list-ec">
                <!---->
                <a class="ecl-description-list__definition-item ecl-link ecl-link--default sc-ecl-description-list-ec" href="/example.html">
                  Lorem ipsum dolor sit amet
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </ecl-link>
              <ecl-link style-class="ecl-description-list__definition-item sc-ecl-description-list-ec">
                <!---->
                <a class="ecl-description-list__definition-item ecl-link ecl-link--default sc-ecl-description-list-ec" href="/example.html">
                  Lorem ipsum dolor sit amet
                  <slot-fb name="icon-after"></slot-fb>
                </a>
              </ecl-link>
            </dd>
          </ecl-description-list-definition>
          <ecl-description-list-term theme="ec">
            <!---->
            <dt class="ecl-description-list__term sc-ecl-description-list-ec">
              Taxonomy list
            </dt>
          </ecl-description-list-term>
          <ecl-description-list-definition items="[&quot;Taxonomy item 1&quot;,{&quot;label&quot;: &quot;Taxonomy item 2&quot;, &quot;path&quot;: &quot;/example.html&quot;},&quot;Taxonomy item 3&quot;]" theme="ec" type="taxonomy">
            <!---->
            <dd class="ecl-description-list__definition ecl-description-list__definition--taxonomy sc-ecl-description-list-ec">
              <span class="ecl-description-list__definition-item sc-ecl-description-list-ec">
                Taxonomy item 1
              </span>
              <span class="ecl-description-list__definition-item sc-ecl-description-list-ec">
                <ecl-link>
                  <!---->
                  <a class="ecl-link ecl-link--default" href="/example.html">
                    Taxonomy item 2
                    <slot-fb name="icon-after"></slot-fb>
                  </a>
                </ecl-link>
              </span>
              <span class="ecl-description-list__definition-item sc-ecl-description-list-ec">
                Taxonomy item 3
              </span>
            </dd>
          </ecl-description-list-definition>
        </dl>
      </ecl-description-list>
    </div>
  </div>
</ecl-content-block>
`;
